contracts/staking/IApplicationWithOperator.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.9;
+
+import "./IApplication.sol";
+
+/// @title  Interface for Threshold Network applications with operator role
+interface IApplicationWithOperator is IApplication {
+    /// @notice Returns operator registered for the given staking provider.
+    function stakingProviderToOperator(address stakingProvider)
+        external
+        view
+        returns (address);
+
+    /// @notice Used by staking provider to set operator address that will
+    ///         operate a node. The operator addressmust be unique.
+    ///         Reverts if the operator is already set for the staking provider
+    ///         or if the operator address is already in use.
+    /// @dev    Depending on application the given staking provider can set operator
+    ///         address only one or multiple times. Besides that application can decide
+    ///         if function reverts if there is a pending authorization decrease for
+    ///         the staking provider.
+    function registerOperator(address operator) external;
+
+    // TODO consider that?
+    // /// @notice Used by additional role (owner for example) to set operator address that will
+    // ///         operate a node for the specified staking provider.
+    // function registerOperator(address stakingProvider, address operator) external;
```

Comment:
What's your opinion on this @vzotova @lukasz-zimnoch ? My suggestion would be to not include it in the interface, so it can be optionally implemented by applications that allow owners to set operators.

Reply:
I'm good with both options

Reply:
> What's your opinion on this @vzotova @lukasz-zimnoch ? My suggestion would be to not include it in the interface, so it can be optionally implemented by applications that allow owners to set operators.

Agreed! That feels optional and should not live on the interface level.

Reply:
Then we need to remove it here as well? https://github.com/threshold-network/token-dashboard/blob/bbdc89047f2f32905cf35ccbb07ac414dc9b0ab2/src/threshold-ts/applications/index.ts#L191

We should probably be using the `solidity-contracts` repo in the dashboard rather than having a bunch of interfaces hard coded there

Reply:
ok, removed
@theref there is "regular" `registerOperator` in your link and this method still in interface

Reply:
oh yes sorry, hadn't noticed the difference in signature.

ok, well the dashboard is using an interface to specify how application contracts should behave. If we don't include the version that allows applications to set Operators then it's not going to be available on the dashboard (unless we completely change the dashboard architecture)